File: samples/client/petstore/java/src/main/java/org/openapijsonschematools/client/schemas/validation/AllOfValidator.java
package org.openapijsonschematools.client.schemas.validation;

import java.util.List;

public class AllOfValidator implements KeywordValidator {
    public final List<Class<? extends JsonSchema>> allOf;

    public AllOfValidator(List<Class<? extends JsonSchema>> allOf) {
        this.allOf = allOf;
    }

    @Override
    public Object getConstraint() {
        return allOf;
    }

    @Override
    public PathToSchemasMap validate(Class<? extends JsonSchema> cls, Object arg, ValidationMetadata validationMetadata, Object extra) {
        PathToSchemasMap pathToSchemas = new PathToSchemasMap();
        for(Class<? extends JsonSchema> allOfClass: allOf) {
            PathToSchemasMap otherPathToSchemas = JsonSchema.validate(allOfClass, arg, validationMetadata);
            pathToSchemas.update(otherPathToSchemas);
        }
        return pathToSchemas;
    }
}

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/AllOfValidator.kt
package org.openapijsonschematools.client.schemas.validation

import org.openapijsonschematools.client.exceptions.ValidationException

class AllOfValidator : KeywordValidator {
    @Throws(ValidationException::class)
    override fun validate(
        data: ValidationData
    ): PathToSchemasMap? {
        val allOf: List<Class<out JsonSchema<*>>> = data.schema.allOf ?: return null
        val pathToSchemas = PathToSchemasMap()
        for (allOfClass in allOf) {
            val allOfSchema = JsonSchemaFactory.getInstance(allOfClass)
            val otherPathToSchemas = JsonSchema.validate(allOfSchema, data.arg, data.validationMetadata)
            pathToSchemas.update(otherPathToSchemas)
        }
        return pathToSchemas
    }
}
